ortools/math_opt/solvers/xpress_solver.cc

```diff
@@ -0,0 +1,693 @@
+// Copyright 2010-2024 Google LLC
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ortools/math_opt/solvers/xpress_solver.h"
+
+#include "absl/strings/str_join.h"
+#include "ortools/base/map_util.h"
+#include "ortools/base/protoutil.h"
+#include "ortools/base/status_macros.h"
+#include "ortools/math_opt/core/math_opt_proto_utils.h"
+#include "ortools/math_opt/core/sparse_vector_view.h"
+#include "ortools/math_opt/cpp/solve_result.h"
+#include "ortools/math_opt/validators/callback_validator.h"
+#include "ortools/port/proto_utils.h"
+#include "ortools/xpress/environment.h"
+
+namespace operations_research {
+namespace math_opt {
+namespace {
+
+absl::Status CheckParameters(const SolveParametersProto& parameters) {
+  std::vector<std::string> warnings;
+  if (parameters.has_threads() && parameters.threads() > 1) {
+    warnings.push_back(absl::StrCat(
+        "XpressSolver only supports parameters.threads = 1; value ",
+        parameters.threads(), " is not supported"));
+  }
```

Comment:
Why is multi-threading not supported? As long as you don't use callbacks, multi-threaded solves are completely transparent to an Xpress user.